src/manager/node.c

```diff
@@ -303,9 +304,10 @@ static int node_match_unit_state_changed(sd_bus_message *m, void *userdata, UNUS
         usubs->substate = strdup(substate);
 
         UnitSubscription *usub = NULL;
-        LIST_FOREACH(subs, usub, usubs->subs) {
+        UnitSubscription *next_usub = NULL;
+        LIST_FOREACH_SAFE(subs, usub, next_usub, usubs->subs) {
```

Comment:
If we change to FOREACH_SAFE here I think it is probably best to do it in all the LIST_FOREACH emits too.